### DISCUSSION TITLE
Is there any way to know when node loading ends?

### DISCUSSION THREAD

**Discussion by tjxtjxtjx** [xyflow/xyflow#4402] [Q&A] [answered]
Hello!

When I use `setNodes(newNodes)` the nodes will change, then I use `fitView(newNodes)` and it doesn't change the view.
If I use 
```
setTimeout(() => {
    fitView(newNodes)
}, 100);
```
Found that the view was correctly changed.

(If the number of nodes in `newNodes` is large, the delay should be increased.)

My guess is that for the view to change correctly you need to wait until the node rendering is finished.

Is there any way to know when the node rendering is finished, I would use `fitView(newNodes)` after the rendering is finished

**Comment by bcakmakoglu** [comment, marked as answer]
There's a hook called [`useNodesInitialized`](https://reactflow.dev/api-reference/hooks/use-nodes-initialized) that you can use for that purpose.

**Reply by tjxtjxtjx** [reply]
Thank you very much! I made a mistake. I thought this hook could only be used when I first opened the canvas!
